recipes/boost/all/conanfile.py

```diff
@@ -59,42 +87,67 @@ class BoostConan(ConanFile):
         "extra_b2_flags": "ANY",  # custom b2 flags
         "i18n_backend": ["iconv", "icu", None],
     }
-    options.update({"without_%s" % libname: [True, False] for libname in lib_list})
+    for _name in CONFIGURE_OPTIONS:
+        options["without_{}".format(_name)] = [True, False]
 
     default_options = {
-        'shared': False,
-        'header_only': False,
-        'error_code_header_only': False,
-        'system_no_deprecated': False,
-        'asio_no_deprecated': False,
-        'filesystem_no_deprecated': False,
-        'fPIC': True,
-        'layout': 'system',
-        'magic_autolink': False,
-        'python_executable': 'None',
-        'python_version': 'None',
-        'namespace': 'boost',
-        'namespace_alias': False,
-        'multithreading': True,
-        'zlib': True,
-        'bzip2': True,
-        'lzma': False,
-        'zstd': False,
-        'segmented_stacks': False,
+        "shared": False,
+        "header_only": False,
+        "error_code_header_only": False,
+        "system_no_deprecated": False,
+        "asio_no_deprecated": False,
+        "filesystem_no_deprecated": False,
+        "fPIC": True,
+        "layout": "system",
+        "magic_autolink": False,
+        "python_executable": "None",
+        "python_version": "None",
```

Comment:
Technical question: Can this be one object? `"python": { "exe": "", "ver": "" }` Does conan support this?

Reply:
Using some fancy syntax, I guess it could be possible.
e.g.:
`-o python:exe="/usr/bin/pyhon3" -o python:ver=3.7`

Reply:
is it supported syntax? I think it conflicts with `-o boost:shared=True`, so `-o python:exe` will apply `exe` option to `python` package.